src/parsers/urdf/model.cpp

```diff
@@ -450,66 +463,15 @@ namespace se3
+        BOOST_FOREACH(::urdf::LinkPtr child, root_link->child_links)
         {
-          typedef std::vector< ::urdf::LinkPtr > LinkSharedPtrVector_t;
-          LinkSharedPtrVector_t movable_child_links;
-          LinkSharedPtrVector_t direct_child_links(root_link->child_links);
-          LinkSharedPtrVector_t next_direct_child_links; // next child to visit
-          LinkSharedPtrVector_t pathologic_child_links; // children which have inertial info but rigidly attached to the world
-          do
-          {
-            next_direct_child_links.clear();
-            BOOST_FOREACH(::urdf::LinkPtr child, direct_child_links)
-            {
-              if (child->parent_joint->type != ::urdf::Joint::FIXED)
-                movable_child_links.push_back(child);
-              else
-              {
-                if (child->inertial)
-                  pathologic_child_links.push_back(child);
-                next_direct_child_links.insert (next_direct_child_links.end(), child->child_links.begin(), child->child_links.end());
-              }
-              
-            }
-            direct_child_links = next_direct_child_links;
-          }
-          while (!direct_child_links.empty());
-          
-          if (!pathologic_child_links.empty())
-          {
-            std::cout << "[INFO] The links:" << std::endl;
-            for (LinkSharedPtrVector_t::iterator it = pathologic_child_links.begin();
-                 it < pathologic_child_links.end(); ++it)
-            {
-              std::cout << "  - " << (*it)->name << std::endl;
-            }
-            std::cout << "are fixed regarding to the universe (base_link) and convey inertial info. They won't affect the dynamics of the output model. Maybe, a root joint is missing connecting this links to the universe." << std::endl;
-            
-          }
-          
-          BOOST_FOREACH(::urdf::LinkPtr child, movable_child_links)
-          {
-            child->getParent()->parent_joint->name = model.names[0];
-            parseTree(child, model, SE3::Identity(), verbose);
-          }
-          
+          parseTree(child, model, SE3::Identity(), verbose);
         }
-        else // Otherwise, we have to rase an exception because the first link will no be added to the model.
-          // It seems a root joint is missing.
-        {
-          std::cout << "[INFO] The root link " << root_link->name << " of the model tree contains inertial information. It seems that a root joint is missing connecting this root link to the universe. The root link won't affect the dynamics of the model." << std::endl;
-          
-          BOOST_FOREACH(::urdf::LinkPtr child, root_link->child_links)
-          {
-            parseTree(child, model, SE3::Identity(), verbose);
-          }
-        }
-        
-        
+
+        // FIXME: check the inertias
+        // The check in appendBodyToJoint only ensures inertias are not NaN
+        // but pinocchio requires non-zero inertias.
```

Comment:
Please replace by: 
// Inertias might be zero due to the URDF file. This is not a bug. However, it would lead to some
// algorithms of Pinocchio not to work. 
// TODO: add an algorithm or a method in model to check the validity of the value wrt to the algorithms the user want to use.
// See also #306 on GitHub.